### PR TITLE
12905: clean up xad extension

### PR DESCRIPTION
**Pull request by mkalish**
This PR collapses all the XAD extensions into one

Test Steps:
1. Tests should cover it all

## Changes
- Updates how the data from the XAD HL7 datatype is stored in a FHIR bundle

### Testing
- [x] Tested locally?
- [x] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [ ] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [ ] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- #12905

### CODE REVIEW

File: prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xtn/XTNTests.kt
Comment (mkalish) [inline prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xtn/XTNTests.kt:1]: I think these are all dead tests or at least covered by the tests in XTNtoContactPoint?  If there is no disagreement I'll delete the file before merging

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml [hunk 1]
```diff
@@ -233,56 +94,40 @@ period_2:
...
+    - expressionType: nested
+      condition: $hl7v2Name NOT_NULL
+      expressionsMap:
+        url:
+          type: SYSTEM_URL
+          value: hl7v2Name
+        valueString:
+          type: STRING
+          valueOf: $hl7v2Name
+    - expressionType: resource
```
Comment (mkalish) [inline prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml:131]: Looking for feedback here.  I opted not to do a null check against field here on the basis of
- something is going to be populated
- if nothing is populated, it just will be a url with an empty `extension` array

File: prime-router/docs/design/design/HL7v2-FHIR-Inventory.md [hunk 2]
```diff
@@ -28,6 +28,26 @@ either store
 HL7 data not translated to FHIR or information on where a particular piece of a FHIR bundle should be mapped to in the
 HL7 message.
 
+### Use of extensions to capture HL7 data
+
+The FHIR bundles that are produced the conversion process are then translated back to HL7 for receivers that want that
```
Comment (brick-green) [inline prime-router/docs/design/design/HL7v2-FHIR-Inventory.md:33]:
```suggestion
The FHIR bundles that are produced via the conversion process are then translated back to HL7 for receivers that want that
```

File: prime-router/docs/design/design/HL7v2-FHIR-Inventory.md [hunk 3]
```diff
@@ -28,6 +28,26 @@ either store
...
+format. In order to ensure:
```
Comment (brick-green) [inline prime-router/docs/design/design/HL7v2-FHIR-Inventory.md:34]: The wording of `In order to ensure:` feels awkward but I can't quite figure out what to recommend instead

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml (hunk 1 above)
Comment (brick-green) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml:131]: I'm not sure what you mean here. Are you talking about a null check on the expression with `iso21090-ADXP-censusTract` for the url?

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/SADExtension.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+url:
+  type: SYSTEM_URL
+  value: sad-address-line
+
+extension:
+  generateList: true
+  expressionType: nested
+  expressions:
+    - expressionType: nested
+      expressionsMap:
```
Comment (brick-green) [inline prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/SADExtension.yml:10]: conditions on these expressions?

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/XADExtension.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+      expressionsMap:
+        url:
+          type: STRING
+          value: XAD.12
+        extension:
+          expressionType: nested
+          generateList: true
+          expressions:
+            - expressionType: nested
+              expressionsMap:
```
Comment (brick-green) [inline prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/XADExtension.yml:84]: conditions on these expressions?

File: prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xtn/XTNTests.kt [hunk 6]
```diff
@@ -3,9 +3,11 @@ package gov.cdc.prime.router.datatests.mappinginventory.xtn
 import gov.cdc.prime.router.datatests.mappinginventory.translateAndCompareFHIRToHL7
 import gov.cdc.prime.router.datatests.mappinginventory.translateAndCompareHL7ToFHIR
 import org.junit.jupiter.api.Test
+import kotlin.test.Ignore
 
 class XTNTests {
     // Tests use (line 2)
+    @Ignore
```
Comment (brick-green) [inline prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xtn/XTNTests.kt:10]: Why do we want to keep and ignore rather than delete?

**Review by brick-green — COMMENTED** [review]
a few comments :)

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml (hunk 1 above)
Comment (mkalish) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml:131]: So if literally none of the values that extension map are present, there will be an empty extension.  i.e. I'm not writing
`xad.1 NULL || xad.2 NULL || xad.3 NULL ...`

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/SADExtension.yml (hunk 4 above)
Comment (mkalish) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/SADExtension.yml:10]: yep will add

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/XADExtension.yml (hunk 5 above)
Comment (mkalish) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/XADExtension.yml:84]: good call

File: prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xtn/XTNTests.kt (hunk 6 above)
Comment (mkalish) [reply prime-router/src/testIntegration/kotlin/datatests/mappinginventory/xtn/XTNTests.kt:10]: Yep will delete before merging

**Comment by mkalish** [comment]
BTY @brick-green

File: prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml (hunk 1 above)
Comment (brick-green) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml:131]: Gotcha. Yes, I think that's okay. Does the XAD.10 expression need a null check?

Comment (mkalish) [reply prime-router/metadata/fhir_mapping/hl7/datatypes/XAD/Address.yml:131]: Yep... missing from the initial implementation.  Will fix it now

**Review by brick-green — APPROVED** [review]
lookin good after the XAD.10 condition

**Comment by mkalish** [comment]
@arnejduranovic Ready for review

**Review by arnejduranovic — APPROVED** [review]

**Pull request merged**